langserver/references.go

```diff
@@ -212,6 +212,32 @@ func (h *LangHandler) handleTextDocumentReferences(ctx context.Context, conn JSO
 		t := time.NewTicker(time.Second)
 		defer t.Stop()
 		streamTick = t.C
+		streamPos := 0
+		streamUpdate = func() {
+			mu.Lock()
+			partial := refs
+			mu.Unlock()
+			if len(partial) == streamPos {
+				return
+			}
+
+			patch := make([]ReferenceAddOp, 0, len(partial)-streamPos)
+			for ; streamPos < len(partial); streamPos++ {
+				patch = append(patch, ReferenceAddOp{
+					OP:    "add",
+					Path:  "/-",
+					Value: goRangeToLSPLocation(fset, partial[streamPos].Pos(), partial[streamPos].End()),
+				})
+			}
+			conn.Notify(ctx, "$/partialResult", &lspext.PartialResultParams{
+				ID: lsp.ID{
```

Comment:
`lsp.ID` is different to `req.ID` which is a `jsonrpc2.ID`. They are actually the same, but I avoided using `jsonrpc2.ID` in `lsp.ID` to avoid adding `jsonrpc2.ID` as a dep. FYI in the version of Go coming out soon I'd just be able to do `ID: lsp.ID(req.ID)`.